metisp/pymetis/src/pymetis/classes/inputs/inputset.py

```diff
@@ -54,6 +57,17 @@ def __init__(self, frameset: cpl.ui.FrameSet, **kwargs):
         self.tag_parameters: dict[str, str] = {}        # A dict of all tunable parameters determined from tags
         self.frameset = frameset
 
+        make_snake = re.compile(r'(?<!^)(?=[A-Z])')
+        cut_input = re.compile(r'Input$')
+
+        # Now iterate over all refined Inputs, instantiate them and feed them the frameset to filter.
+        for (name, input_type) in inspect.getmembers(self.__class__,
+                                                     lambda x: inspect.isclass(x) and issubclass(x, PipelineInput)):
+            inp = input_type(frameset)
+            # FixMe: very hacky for now: determine the name of the instance from the name of the class
```

Comment:
Just run with it! I can understand it, I think:

`(?<!^)(?=[A-Z])` means: look-back to determine we are not at the start, and then look-here/ahead but don't consume (?) and this is a capital. If so, insert an underscore. Before all that, remove "Input", and afterwards make lowercase. So this is debugable.